### PR TITLE
Clear existing children when connecting subqueries to plan

### PR DESCRIPTION
**Pull request by wolffcm**
When I was writing some planner tests, I was encountering failed
assertions because a scan plan node had more than one child.  I
traced the cause to the method `connectChildrenBestPlans` which seems
to have been added as part of in/exists/scalar subquery support.

I wasn't able to recreate this issue using sqlcmd (I tried both ad hoc
and procedures), but seems like this issue might manifest in a
real-world scenario?

The added test was failing assertions in the planner before the fix.

### CODE REVIEW

File: src/frontend/org/voltdb/planner/PlanAssembler.java [hunk 1]
```diff
@@ -812,6 +812,7 @@ private AbstractPlanNode connectChildrenBestPlans(AbstractPlanNode parentPlan) {
                 assert (betsCostPlan != null);
                 AbstractPlanNode subQueryRoot = betsCostPlan.rootPlanGraph;
                 subQueryRoot.disconnectParents();
```
Comment (BillWhite) [inline src/frontend/org/voltdb/planner/PlanAssembler.java]: While we are about it, should this be bestCostPlan, not betsCostPlan?

**Comment by BillWhite** [comment]
With one typo comment, this seems right to me.  I'm not 100% sure I understand it, though.

**Comment by paulmartel** [comment]
Looks like progress to me. Good to go.

**Comment by wolffcm** [comment]
Thanks guys.  Something weird is happening in Jenkins, but I'll merge once it looks good.

**Pull request merged**